Transient prompt stops working in fish when using fish vi or hybrid key bindings

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

Transient prompt is drawn correctly only when using `fish_default_key_bindings`. After running `fish_hybrid_key_bindings` or `fish_vi_key_bindings`, transient prompt doesn't work. Regular prompt is printed each time. After running `fish_default_key_bindings` transient prompt gets back to normal.

### Theme

```
$schema: https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json
version: 2
console_title_template: '{{if .Root}}(Admin){{end}} {{.PWD}}'
transient_prompt:
  background: "transparent"
  foreground: "green"
  foreground_templates:
    - '{{ if gt .Code 0 }}red{{ end }}'
  template: '{{ if gt .Code 0 }}  {{else}}  {{ end }}'
blocks:
  - type: prompt
    alignment: left
    segments:
      - type: exit
        style: plain
        foreground: green
        foreground_templates:
          - '{{ if gt .Code 0 }}red{{ end }}'
        template: '│'
        properties:
          always_enabled: true
      - type: path
        style: plain
        foreground: blue
        template: '  {{ .Path }}'
        properties:
          style: full
      - type: git
        style: plain
        foreground: bright-black
        template: ' {{ .HEAD }}{{if .BranchStatus }} {{ .BranchStatus }}{{ end }}{{
          if .Working.Changed }}  {{ .Working.String }}{{ end }}{{ if and (.Staging.Changed)
          (.Working.Changed) }} |{{ end }}{{ if .Staging.Changed }}  {{ .Staging.String
          }}{{ end }}'
        properties:
          branch_ahead_icon: <cyan>⇡ </>
          branch_behind_icon: <cyan>⇣ </>
          branch_icon: ' '
          commit_icon: ' '
          tag_icon: ' '
          fetch_stash_count: true
          fetch_status: true
          fetch_bare_info: true
      - type: kubectl
        style: plain
        foreground: yellow
        template: ' 󱃾 {{ .Cluster }}'
      - type: executiontime
        style: plain
        foreground: red
        template: '  {{ .FormattedMs }}'
        properties:
          style: round
  - type: prompt
    newline: true
    alignment: left
    segments:
      - type: exit
        style: plain
        foreground: green
        foreground_templates:
          - '{{ if gt .Code 0 }}red{{ end }}'
        template: '│ '
        properties:
          always_enabled: true
```

### What OS are you seeing the problem on?

Linux

### Which shell are you using?

fish

### Log output

```Shell
Version: 14.17.0

Segments:

ConsoleTitle(true)                         -   1 ms -  ~
exit(true)                                 -   0 ms - │
path(true)                                 -   0 ms -   ~
git(false)                                 -   3 ms - 
kubectl(true)                              -  58 ms -  󱃾 mycluster-mydomain
executiontime(false)                       -   0 ms - 
exit(true)                                 -   0 ms - │ 

Run duration: 73.002112ms

Cache path: /home/user/.cache/oh-my-posh

Config path: /home/user/.dotfiles/oh-my-posh/vertical.omp.yaml

Logs:

[DEBUG] 11:41:52.188 shell.go:Getenv:392 → tmux
[TRACE] 11:41:52.188 shell.go:Getenv(TERM_PROGRAM) - 28.294µs
[DEBUG] 11:41:52.188 shell.go:Getenv:392 → NO DATA
[TRACE] 11:41:52.189 shell.go:Getenv(XDG_CACHE_HOME) - 3.346µs
[TRACE] 11:41:52.189 shell_unix.go:CachePath() - 35.161µs
[DEBUG] 11:41:52.192 shell.go:Getenv:392 → /home/user/.dotfiles/oh-my-posh/vertical.omp.yaml
[TRACE] 11:41:52.192 shell.go:Getenv(POSH_THEME) - 13.059µs
[DEBUG] 11:41:52.192 shell_unix.go:Platform:63 → manjaro
[TRACE] 11:41:52.192 shell.go:resolveConfigPath() - 24.156µs
[TRACE] 11:41:52.192 shell.go:Init() - 3.292782ms
[TRACE] 11:41:52.192 shell.go:Flags() - 67ns
[TRACE] 11:41:52.196 config.go:loadConfig() - 4.506254ms
[TRACE] 11:41:52.196 shell.go:Flags() - 151ns
[DEBUG] 11:41:52.196 shell.go:Getenv:392 → NO DATA
[TRACE] 11:41:52.196 shell.go:Getenv(OMP_CACHE_DISABLED) - 8.012µs
[TRACE] 11:41:52.197 shell_unix.go:Root() - 984ns
[DEBUG] 11:41:52.198 shell.go:Shell:641 → process name: fish
[TRACE] 11:41:52.198 shell.go:Shell() - 1.030328ms
[TRACE] 11:41:52.198 shell.go:ErrorCode() - 66ns
[DEBUG] 11:41:52.198 shell.go:Getenv:392 → NO DATA
[TRACE] 11:41:52.198 shell.go:Getenv(WSL_DISTRO_NAME) - 3.989µs
[TRACE] 11:41:52.198 shell_unix.go:IsWsl() - 7.334µs
[TRACE] 11:41:52.198 shell.go:GOOS() - 91ns
[DEBUG] 11:41:52.198 shell.go:Pwd:420 → NO DATA
[TRACE] 11:41:52.198 shell.go:Pwd() - 12.691µs
[TRACE] 11:41:52.198 shell.go:GOOS() - 62ns
[TRACE] 11:41:52.198 shell.go:GOOS() - 39ns
[TRACE] 11:41:52.198 shell.go:GOOS() - 38ns
[TRACE] 11:41:52.198 shell.go:GOOS() - 39ns
[TRACE] 11:41:52.198 shell.go:GOOS() - 61ns
[TRACE] 11:41:52.198 shell.go:GOOS() - 59ns
[TRACE] 11:41:52.198 shell.go:GOOS() - 40ns
[TRACE] 11:41:52.198 shell.go:GOOS() - 39ns
[TRACE] 11:41:52.198 shell.go:GOOS() - 39ns
[DEBUG] 11:41:52.198 shell.go:User:545 → user
[TRACE] 11:41:52.198 shell.go:User() - 3.91µs
[DEBUG] 11:41:52.198 shell.go:Host:557 → lwro1016x
[TRACE] 11:41:52.198 shell.go:Host() - 12.398µs
[TRACE] 11:41:52.198 shell.go:GOOS() - 62ns
[DEBUG] 11:41:52.198 shell_unix.go:Platform:63 → manjaro
[TRACE] 11:41:52.198 shell.go:TemplateCache() - 1.18208ms
[TRACE] 11:41:52.198 shell.go:Flags() - 65ns
[DEBUG] 11:41:52.198 shell.go:Pwd:400 → /home/user
[TRACE] 11:41:52.198 shell.go:Pwd() - 4.236µs
[TRACE] 11:41:52.198 shell.go:GOOS() - 61ns
[TRACE] 11:41:52.198 shell.go:ErrorCode() - 41ns
[TRACE] 11:41:52.198 shell.go:TemplateCache() - 117ns
[TRACE] 11:41:52.198 shell.go:Shell() - 111ns
[TRACE] 11:41:52.198 shell.go:TemplateCache() - 163ns
[DEBUG] 11:41:52.198 shell.go:Pwd:400 → /home/user
[TRACE] 11:41:52.198 shell.go:Pwd() - 6.85µs
[TRACE] 11:41:52.198 shell.go:GOOS() - 87ns
[DEBUG] 11:41:52.198 shell.go:Pwd:400 → /home/user
[TRACE] 11:41:52.198 shell.go:Pwd() - 3.407µs
[TRACE] 11:41:52.198 shell.go:Shell() - 109ns
[TRACE] 11:41:52.198 shell.go:Shell() - 43ns
[TRACE] 11:41:52.198 shell.go:GOOS() - 64ns
[TRACE] 11:41:52.198 shell.go:PathSeparator() - 110ns
[TRACE] 11:41:52.198 shell.go:GOOS() - 61ns
[TRACE] 11:41:52.198 shell.go:GOOS() - 38ns
[TRACE] 11:41:52.198 shell.go:GOOS() - 53ns
[TRACE] 11:41:52.198 shell.go:GOOS() - 76ns
[TRACE] 11:41:52.198 shell.go:PathSeparator() - 46ns
[TRACE] 11:41:52.198 shell.go:PathSeparator() - 38ns
[TRACE] 11:41:52.198 shell.go:GOOS() - 61ns
[TRACE] 11:41:52.198 shell.go:GOOS() - 39ns
[TRACE] 11:41:52.198 shell.go:PathSeparator() - 61ns
[TRACE] 11:41:52.198 shell.go:GOOS() - 39ns
[TRACE] 11:41:52.198 shell.go:GOOS() - 38ns
[TRACE] 11:41:52.198 shell.go:PathSeparator() - 61ns
[TRACE] 11:41:52.198 shell.go:GOOS() - 41ns
[TRACE] 11:41:52.199 shell.go:GOOS() - 38ns
[TRACE] 11:41:52.199 shell.go:PathSeparator() - 62ns
[DEBUG] 11:41:52.199 shell.go:Pwd:400 → /home/user
[TRACE] 11:41:52.199 shell.go:Pwd() - 3.732µs
[DEBUG] 11:41:52.199 shell.go:Getenv:392 → NO DATA
[TRACE] 11:41:52.199 shell.go:Getenv(WSL_DISTRO_NAME) - 2.977µs
[TRACE] 11:41:52.199 shell_unix.go:IsWsl() - 6.941µs
[TRACE] 11:41:52.199 shell.go:StackCount() - 101ns
[TRACE] 11:41:52.199 shell_unix.go:DirIsWritable(/home/user) - 8.539µs
[TRACE] 11:41:52.199 shell.go:TemplateCache() - 103ns
[TRACE] 11:41:52.199 shell.go:TemplateCache() - 158ns
[TRACE] 11:41:52.199 shell.go:Shell() - 62ns
[DEBUG] 11:41:52.199 shell.go:Pwd:400 → /home/user
[TRACE] 11:41:52.199 shell.go:Pwd() - 4.534µs
[TRACE] 11:41:52.199 shell.go:GOOS() - 109ns
[DEBUG] 11:41:52.199 shell.go:Getenv:392 → NO DATA
[TRACE] 11:41:52.199 shell.go:Getenv(WSL_DISTRO_NAME) - 2.956µs
[TRACE] 11:41:52.199 shell_unix.go:IsWsl() - 6.064µs
[TRACE] 11:41:52.199 shell_unix.go:IsWsl2() - 9.46µs
[TRACE] 11:41:52.199 shell.go:GOOS() - 58ns
[DEBUG] 11:41:52.199 shell.go:CommandPath:596 → /usr/bin/git
[TRACE] 11:41:52.199 shell.go:CommandPath(git) - 36.792µs
[TRACE] 11:41:52.199 shell.go:HasCommand(git) - 40.633µs
[DEBUG] 11:41:52.199 shell.go:Pwd:400 → /home/user
[TRACE] 11:41:52.199 shell.go:Pwd() - 3.544µs
[ERROR] 11:41:52.199 shell.go:HasParentFilePath:729 → stat .git: no such file or directory
[TRACE] 11:41:52.199 shell.go:HasParentFilePath(.git) - 26.666µs
[DEBUG] 11:41:52.199 shell.go:Pwd:400 → /home/user
[TRACE] 11:41:52.199 shell.go:Pwd() - 4.76µs
[ERROR] 11:41:52.202 shell.go:RunCommand:573 → exit status 128
[DEBUG] 11:41:52.202 shell.go:RunCommand:575 ↓
    fatal: to nie jest repozytorium gita (ani żaden z katalogów nadrzędnych): .git
    
[TRACE] 11:41:52.202 shell.go:RunCommand(git -C /home/user --no-optional-locks -c core.quotepath=false -c color.status=false rev-parse --is-bare-repository) - 3.532553ms
[DEBUG] 11:41:52.202 shell.go:Pwd:400 → /home/user
[TRACE] 11:41:52.202 shell.go:Pwd() - 5.484µs
[TRACE] 11:41:52.202 shell.go:GOOS() - 72ns
[DEBUG] 11:41:52.203 shell.go:CommandPath:596 → /usr/bin/kubectl
[TRACE] 11:41:52.203 shell.go:CommandPath(kubectl) - 29.807µs
[TRACE] 11:41:52.203 shell.go:HasCommand(kubectl) - 33.609µs
[DEBUG] 11:41:52.260 shell.go:RunCommand:575 ↓
    apiVersion: v1
    clusters:
    - cluster:
        certificate-authority-data: DATA+OMITTED
        server: https://mycluster-m1.mydomain:6443
      name: mycluster-mydomain
    contexts:
    - context:
        cluster: mycluster-mydomain
        user: admin-mycluster-mydomain
      name: admin@mycluster-mydomain
    current-context: admin@mycluster-mydomain
    kind: Config
    preferences: {}
    users:
    - name: admin-mycluster-mydomain
      user:
        client-certificate-data: DATA+OMITTED
        client-key-data: DATA+OMITTED
[TRACE] 11:41:52.260 shell.go:RunCommand(kubectl config view --output yaml --minify) - 57.637561ms
[TRACE] 11:41:52.260 shell.go:TemplateCache() - 359ns
[TRACE] 11:41:52.260 shell.go:TemplateCache() - 122ns
[TRACE] 11:41:52.261 shell.go:Shell() - 266ns
[DEBUG] 11:41:52.261 shell.go:Pwd:400 → /home/user
[TRACE] 11:41:52.261 shell.go:Pwd() - 6.052µs
[TRACE] 11:41:52.261 shell.go:GOOS() - 64ns
[TRACE] 11:41:52.261 shell.go:ExecutionTime() - 65ns
[TRACE] 11:41:52.261 shell.go:Flags() - 53ns
[DEBUG] 11:41:52.261 shell.go:Pwd:400 → /home/user
[TRACE] 11:41:52.261 shell.go:Pwd() - 2.953µs
[TRACE] 11:41:52.261 shell.go:GOOS() - 45ns
[TRACE] 11:41:52.261 shell.go:ErrorCode() - 52ns
[TRACE] 11:41:52.261 shell.go:TemplateCache() - 91ns
[TRACE] 11:41:52.261 shell.go:Shell() - 38ns
[TRACE] 11:41:52.261 shell.go:TemplateCache() - 313ns
[DEBUG] 11:41:52.261 shell.go:Getenv:392 → NO DATA
[TRACE] 11:41:52.261 shell.go:Getenv(XDG_CACHE_HOME) - 7.36µs
[TRACE] 11:41:52.262 shell_unix.go:CachePath() - 25.683µs
[TRACE] 11:41:52.262 shell.go:Flags() - 109ns
```

--- a/src/shell/scripts/omp.fish
+++ b/src/shell/scripts/omp.fish
@@ -100,7 +100,8 @@ function _render_tooltip
 end
 
 if test "::TOOLTIPS::" = "true"
-  bind \x20 _render_tooltip
+  bind \x20 _render_tooltip -M default
+  bind \x20 _render_tooltip -M insert
 end
 
 # transient prompt
@@ -116,7 +117,9 @@ function _render_transient
 end
 
 if test "::TRANSIENT::" = "true"
-  bind \r _render_transient
+  bind \r _render_transient -M default
+  bind \r _render_transient -M insert
+  bind \r _render_transient -M visual
 end
 
 # legacy functions

